addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
@toranb runs are intended to be synchronous, so by having an ajax request return asynchronously the run which triggered it should already be complete.